### PR TITLE
[DependencyInjection] Remove static from closures

### PR DESCRIPTION
**Pull request by ruudk**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  |no 
| Deprecations? |no 
| Tickets       | 
| License       | MIT
| Doc PR        | 

As per request from @nicolas-grekas.

https://github.com/symfony/symfony/pull/49472#issuecomment-1438595345

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Thank you @ruudk.

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -244,7 +244,7 @@ public function dump(array $options = []): string|array
         if ($this->addGetService) {
             $code = preg_replace(
                 "/(\r?\n\r?\n    public function __construct.+?\\{\r?\n) ++([^\r\n]++)/s",
-                "\n    protected \Closure \$getService;$1        \$containerRef = $2\n        \$this->getService = static function () use (\$containerRef) { return \$containerRef->get()->getService(...\\func_get_args()); };",
+                "\n    protected \Closure \$getService;$1        \$containerRef = $2\n        \$this->getService = function () use (\$containerRef) { return \$containerRef->get()->getService(...\\func_get_args()); };",
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php:247]: this one does not look like a closure being in a static method, as it has `$this->getService = `

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 2]
```diff
@@ -1545,7 +1545,7 @@ private function addInlineRequires(bool $hasProxyClasses): string
             $code .= "\n            include_once __DIR__.'/proxy-classes.php';";
         }
 
-        return $code ? sprintf("\n        \$this->privates['service_container'] = static function (\$container) {%s\n        };\n", $code) : '';
+        return $code ? sprintf("\n        \$this->privates['service_container'] = function (\$container) {%s\n        };\n", $code) : '';
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php:1548]: same here

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php (hunk 1 above)
Comment (ruudk) [reply src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php:247]: @nicolas-grekas Do I need to create a PR to fix this?

Comment (ruudk) [reply src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php:247]: If it's needed, here it is: https://github.com/symfony/symfony/pull/49476

**Pull request merged**